app/ts/MigrationUI/components/Migration.tsx

```diff
@@ -0,0 +1,113 @@
+			const parsedExtraInfo = getParsedExtraInfo(newMarketData.extraInfo)
+			forkingMarketData.deepValue = { marketAddress: universeForkingInformation.deepValue.forkingMarket, parsedExtraInfo, hotLoadingMarketData: newMarketData }
+
+			const marketType = MARKET_TYPES[forkingMarketData.deepValue.hotLoadingMarketData.marketType]
+			if (marketType === undefined) throw new Error('invalid marketType')
+			forkingoutcomeStakes.deepValue = getOutcomeNamesAndNumeratorCombinationsForMarket(marketType, forkingMarketData.deepValue.hotLoadingMarketData.numOutcomes, forkingMarketData.deepValue.hotLoadingMarketData.numTicks, forkingMarketData.deepValue.hotLoadingMarketData.outcomes)
+		}
+	}
+
+	const migrateReputationToChildUniverseByPayoutButton = async () => {
+		const account = maybeAccountAddress.peek()
+		if (account === undefined) throw new Error('missing maybeAccountAddress')
+		if (reputationTokenAddress.deepValue === undefined) throw new Error('missing reputationTokenAddress')
+		if (forkingoutcomeStakes.deepValue === undefined) throw new Error('missing forkingoutcomeStakes')
+		const payoutNumerators = forkingoutcomeStakes.deepValue.find((outcome) => outcome.outcomeName === selectedOutcome.value)?.payoutNumerators
+		if (!payoutNumerators) throw new Error('Selected outcome not found')
+		await migrateReputationToChildUniverseByPayout(account.value, reputationTokenAddress.deepValue, payoutNumerators, repV2ToMigrate.value)
+	}
+
+	const migrateFromRepV1toRepV2GenesisTokenButton = async () => {
+		const account = maybeAccountAddress.peek()
+		if (account === undefined) throw new Error('missing maybeAccountAddress')
+		await migrateFromRepV1toRepV2GenesisToken(account.value)
+	}
+
+	const approveRepV1ForMigration = async () => {
+		const account = maybeAccountAddress.peek()
+		if (account === undefined) throw new Error('missing maybeAccountAddress')
+		if (v1ReputationBalance.deepValue === undefined) throw new Error('missing v1ReputationBalance balance')
+		await approveErc20Token(account.value, REPUTATION_V1_TOKEN_ADDRESS, REPV2_TOKEN_ADDRESS, v1ReputationBalance.deepValue)
+	}
+
+	if (universe.deepValue === undefined || reputationTokenAddress.deepValue === undefined || universeForkingInformation.deepValue === undefined) return <></>
+	return <div class = 'subApplication'>
+		<button class = 'button is-primary' onClick = { update }>Update data</button>
+		<div class = 'panel'>
+			<div style = 'display: grid'>
+				<span><b>Universe Name:</b>{ getUniverseName(universe.deepValue) }</span>
+				<span><b>Universe Address:</b>{ universe.deepValue }</span>
+				<span><b>Reputation V2 Address For The Universe:</b>{ universe.deepValue }</span>
+				<span><b>Is Universe Forking:</b>{ universeForkingInformation.deepValue.isForking ? 'Yes' : 'No' }</span>
+				<span><b>Forking End Time:</b>{ universeForkingInformation.deepValue.forkEndTime === undefined ? 'Not Forking' : formatUnixTimestampISO(universeForkingInformation.deepValue.forkEndTime) }</span>
```

Comment:
Your screenshot has these blank rather than populated.  Is the screenshot from before fixing some bug, or is this code incorrect?

Reply:
the screenshot is before fixing those